evalml/pipelines/components/transformers/preprocessing/drop_nan_rows_transformer.py

```diff
@@ -43,12 +44,25 @@ def transform(self, X, y=None):
         y_t = infer_feature_types(y) if y is not None else None
 
         X_t_schema = X_t.ww.schema
+        y_t_logical = None
+        y_t_semantic = None
         if y_t is not None:
-            y_t_logical = y_t.ww.logical_type
-            y_t_semantic = y_t.ww.semantic_tags
+            if isinstance(y_t, pd.DataFrame):
+                y_t_logical = y_t.ww.logical_types
+                y_t_semantic = y_t.ww.semantic_tags
+            else:
+                y_t_logical = y_t.ww.logical_type
+                y_t_semantic = y_t.ww.semantic_tags
```

Comment:
Super tiny nit: we can move `y_semantic` out of the logical branches since the call is the same for both